packages/core-snapshots/src/workers/worker-wrapper.ts

```diff
@@ -0,0 +1,96 @@
+import { EventEmitter } from "events";
+import { Worker } from "worker_threads";
+
+export class WorkerWrapper extends EventEmitter {
+    private worker: Worker;
+    private isDone: boolean = false;
+
+    public constructor(data: any) {
+        super();
+        this.worker = new Worker(__dirname + "/worker.js", { workerData: data });
+
+        this.worker.on("message", (data) => {
+            // console.log("Message", data);
+
+            this.handleMessage(data);
+        });
+
+        this.worker.on("error", (err) => {
+            // this.emit("error", err)
+            this.emit("*", { name: "error", data: err });
+        });
+
+        // this.worker.on("error", err => {
+        //     this.emit("error", err)
+        //     this.emit("*", { name: "error", data: err })
+        // })
+
+        this.worker.on("exit", (statusCode) => {
+            this.isDone = true;
+            this.emit("exit", statusCode);
+            this.emit("*", { name: "exit", data: statusCode });
+        });
+    }
+
+    public start(): Promise<void> {
+        return new Promise((resolve, reject) => {
+            this.once("*", (data) => {
+                if (data.name === "started") {
+                    resolve();
+                } else if (data.name === "exit") {
+                    resolve();
+                } else if (data.name === "exception" || data.name === "error") {
+                    reject(data.data);
+                } else {
+                    reject();
+                }
+            });
+
+            this.worker.postMessage({ action: "start" });
+        });
+    }
+
+    public sync(data: any): Promise<any> {
+        return new Promise((resolve, reject) => {
+            if (this.isDone) {
+                /* istanbul ignore next */
+                resolve();
+                /* istanbul ignore next */
+                return;
+            }
+
+            this.once("*", (data) => {
+                if (data.name === "synchronized") {
+                    resolve(data.data);
+                } else if (data.name === "exit") {
+                    resolve();
+                } else if (data.name === "exception" || data.name === "error") {
+                    reject(data.data);
+                } else {
+                    reject();
+                }
+            });
+
+            this.worker.postMessage({
+                action: "sync",
+                data: data,
+            });
+        });
+    }
+
+    public async terminate() {
+        await this.worker.terminate();
+    }
+
+    private handleMessage(data) {
+        // if (data.action === "log") {
```

Comment:
Is this unused or will this be implemented to be something?

Reply:
It is not implemented. I used it for debugging worker and I left it commented as reminder if someone wants to log things from worker. Now it is removed.